x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/hooks/export_csv.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
+import type { EsQuerySortValue, SearchSourceFields } from '@kbn/data-plugin/common';
+import { DataView, SortDirection } from '@kbn/data-plugin/common';
+import { ReportingAPIClient } from '@kbn/reporting-public';
+import { toMountPoint } from '@kbn/react-kibana-mount';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { FieldFormatsStartCommon } from '@kbn/field-formats-plugin/common';
+
+import {
+  sendGetAgentStatusRuntimeField,
+  useKibanaVersion,
+  useStartServices,
+} from '../../../../../../hooks';
+import type { Agent } from '../../../../../../../common';
+import { getSortConfig, removeSOAttributes } from '../../../../../../../common';
+
+import { getSortFieldForAPI } from './use_fetch_agents_data';
+
+export function useExportCSV(enableExportCSV?: boolean) {
+  const startServices = useStartServices();
+  const { notifications, http, uiSettings } = startServices;
+  const kibanaVersion = useKibanaVersion();
+  const [runtimeFields, setRuntimeFields] = useState<string>('emit("online")');
+
+  useEffect(() => {
```

Comment:
is there a way to use a `useQuery(()` here instead of that useEffect?

Reply:
done